### ISSUE TITLE
Can the

### ISSUE THREAD

**Issue by KazuhiraDZ** [dasguptar/treelstm.pytorch#16]
I run the sentiment model successfully. My gpus are double 1080ti, and get a 14%  in gpu0. Is there an extra way to run it on multigpu? I implement a model in tensorflow fold, but it seems that it can't support multigpu.

**Comment by dasguptar** [comment]
Hi @KazuhiraDZ , I have not implemented the sentiment model here. Have you modified the code to work on the sentiment model?

In terms of MultiGPU, I think the better way would be to use PyTorch multiprocessing. These links can get you started:
- https://github.com/pytorch/examples/tree/master/mnist_hogwild
- http://pytorch.org/docs/master/multiprocessing.html

The reason I suggest multiprocessing over using PyTorch MultiGPU is that my implementation of TreeLSTM operates at a batch size of `1`. This means very little GPU memory is utilized, and the CPU version is just as fast because there is no batching. Considering this, running multiple models in parallel and averaging the gradients using multiprocessing seems to be a better way to improve throughput, at least to me.

**Comment by KazuhiraDZ** [comment]
@dasguptar Thank you for advice! I consider transforming my sentiment model to pytorch. As I used tensorflow before, these links you give may help me to start pytorch. And, is that a way to expand the batch size of your model just like batch processing, I means train n trees once time. So I can utilize my several gpus.(I am afraid my cpu can run so fast)
